File: boot-data-redis/src/main/java/com/example/demo/PostRepository.java
package com.example.demo;

import java.util.UUID;
import org.springframework.data.redis.core.ReactiveRedisOperations;
import org.springframework.stereotype.Component;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Component
class PostRepository {

  ReactiveRedisOperations<String, Post> template;

  public PostRepository(ReactiveRedisOperations<String, Post> template) {
    this.template = template;
  }

  Flux<Post> findAll() {
    return template.<String, Post>opsForHash().values("posts");
  }

  Mono<Post> findById(String id) {
    return template.<String, Post>opsForHash().get("posts", id);
  }

  Mono<Post> save(Post post) {
    if (post.getId() != null) {
      String id = UUID.randomUUID().toString();
      post.setId(id);
    }
    return template.<String, Post>opsForHash().put("posts", post.getId(), post)
        .log()
        .map(p -> post);

  }

  Mono<Void> deleteById(String id) {
    return template.<String, Post>opsForHash().remove("posts", id)
        .flatMap(p -> Mono.<Void>empty());
  }

  Mono<Boolean> deleteAll() {
    return template.<String, Post>opsForHash().delete("posts");
  }

}

File: kotlin-gradle/src/main/kotlin/com/example/demo/PostRepository.kt
package com.example.demo

import org.springframework.data.mongodb.core.ReactiveMongoTemplate
import org.springframework.data.mongodb.core.query.Criteria
import org.springframework.data.mongodb.core.query.Query
import org.springframework.data.mongodb.repository.ReactiveMongoRepository

class PostRepository(private val template: ReactiveMongoTemplate) {
    fun findAll() = template.findAll(Post::class.java)
    fun findById(id: String) = template.findById(id, Post::class.java)
    fun save(post: Post) = template.save(post)
    fun deleteAll() = template.remove(Query(), Post::class.java)
    fun deleteById(id: String) = template.remove(Query().addCriteria(Criteria.where("id").`is`(id)), Post::class.java)
}